25 monsterattackhandler

Y1Egger opened the pull request:
Monster can now damage the tower.

Y1Egger committed:
feat: Add MonsterAttackDTO class to dto package

Created the MonsterAttackDTO class in the "dto" package to represent data transfer objects related to monster attacks.

Files added:
- Added MonsterAttackDTO.java to dto package

Y1Egger committed:
feat: Add MAHService class to service package

Created the MAHService class in the "service" package to manage and handle specific functionality related to the application's services.

Files modified:
- Added MAHService.java to service package

RenaT0Pl committed:
feat: Added Constructor to MonsterAttackDTO

Y1Egger committed:
feat: Override abstract method in MonsterAttackDTO

Files modified:
- Modified MonsterAttackDTO.java

Y1Egger committed:
feat: Add functionality to MHAService

Files modified:
- Modified MHAService.java

RenaT0Pl committed:
feat: Updated Functionality of MAHService

RenaT0Pl committed:
feat: Added handleMessage Functionality

Y1Egger committed:
feat: Updated MonsterAttackHandler

Files modified:
- Modified MonsterAttackHandler.java

RenaT0Pl committed:
feat: Added MAHServiceTest

RenaT0Pl committed:
feat: Added MonsterAttackDTOTest

RenaT0Pl committed:
feat: Added MAHServiceTest Functionality

Y1Egger committed:
refactor: MAHServiceTest.java

Y1Egger committed:
feat: Updated MAHServiceTest.java

Y1Egger committed:
feat: Added functionality to MonsterAttackDTOTest.java

RenaT0Pl committed:
refactor: Refactoring MonsterAttackDTOTest.java

RenaT0Pl committed:
feat: Added Functionality to "MonsterAttackDTOTest.java"

Y1Egger committed:
refactor: MonsterAttackDTOTest set access modifier to protected

Y1Egger committed:
refactor: MAHServiceTest set access modifier to protected

RenaT0Pl committed:
refactor: Used methods from UtilityMethods instead of Logger

RenaT0Pl committed:
refactor: Removed smells

Y1Egger committed:
refactor: Updated MonsterAttackHandler

RenaT0Pl committed:
Merge remote-tracking branch 'origin/25-monsterattackhandler' into 25-monsterattackhandler

RenaT0Pl committed:
feat: Updated MonsterAttackHandler

Y1Egger committed:
refactor: optimized classes

Y1Egger committed:
refactor: adjusted classes to work with the client

RenaT0Pl committed:
chore: Updated TowerImpl for Sprint 2

For the beginning, we will have 1 big tower with 10 hp instead of 4 walls and 4 towers.

RenaT0Pl committed:
Merge remote-tracking branch 'origin/25-monsterattackhandler' into 25-monsterattackhandler

paulxyh commented on .idea/misc.xml:
Line 1 wieder einfügen => Könnte zu einem Projekterror führen

Y1Egger replied:
Line 1 wurde wieder hinzugefügt :)

paulxyh commented on src/main/java/at/aau/se2/handler/game/subhandlers/MonsterAttackHandler.java:
```diff
@@ -1,16 +1,78 @@
+        }
+    }
+
+    private boolean isMessageTypeValid(JsonNode msg) {
```
Gehört in MAHService

RenaT0Pl replied:
Wurde nach MAHService verschoben.

paulxyh commented on src/main/java/at/aau/se2/handler/game/subhandlers/MonsterAttackHandler.java:
```diff
@@ -1,16 +1,78 @@
+        return "MONSTER_ATTACK".equals(msg.path("type").asText());
+    }
+
+    private boolean validateEntities(String monsterId, String towerId, Lobby lobby) {
```
Gehört in MAHService

RenaT0Pl replied:
Wurde nach MAHService verschoben.

paulxyh commented on src/main/java/at/aau/se2/handler/game/subhandlers/MonsterAttackHandler.java:
```diff
@@ -1,16 +1,78 @@
+        triggerMonsterAttack(monsterId, towerId, lobby);
+        MonsterAttackDTO dto = buildAttackDTO(monsterId, towerId, lobby);
+        sendMessage(session, dto);
+    }
```
Gehört in MAHservice

RenaT0Pl replied:
Wurde größtenteils in der Form entfernt.

paulxyh commented on src/main/java/at/aau/se2/handler/game/subhandlers/MonsterAttackHandler.java:
```diff
@@ -1,16 +1,78 @@
+        } catch (Exception e) {
+            logs("Error sending message to client: " + e.getMessage());
+        }
+    }
```
Bitte den `sendMessage` Aufruf in die `handleMessage` Methode

Y1Egger replied:
Der sendMessage-Aufruf ist nun in der handleMessage-Methode.

paulxyh commented on src/main/java/at/aau/se2/handler/game/subhandlers/MonsterAttackHandler.java:
```diff
@@ -1,16 +1,78 @@
+        }
+    }
 
+    private MonsterAttackDTO buildAttackDTO(String monsterId, String towerId, Lobby lobby) {
```
in der `handleMessage` Methode machen => nur Konstruktion-Aufruf der MonsterAttackDTO Klasse

Y1Egger replied:
Konstruktions-Aufruf wird nun in der HandleMessage Methode gemacht.

paulxyh commented on src/main/java/at/aau/se2/handler/game/subhandlers/MonsterAttackHandler.java:
...Handler - Klassen sollen nur die `handleMessage` Methode enthalten

Y1Egger replied:
Die Handler-Klasse enthält nur noch handleMessage Methode.

Y1Egger committed:
fix: adjusted misc.xml accordingly

RenaT0Pl committed:
chore: Updated TowerImplTest for Sprint 2

For the beginning, we will have 1 big tower with 10 hp instead of 4 walls and 4 towers.

RenaT0Pl committed:
refactor: Moved several helper methods from MonsterAttackHandler to MAHService

Modified file: MAHService

Y1Egger committed:
feat: Added additional test cases in MAHServiceTest for 100% coverage

Added tests cases for helper-methods:
- isMessageTypeValid
- validateEntities
- findMonsterById
- findTowerById
- and both conditions in "if (tower != null && monster != null)"

Y1Egger committed:
refactor: Changed MonsterAttackHandler according to the review

- Only handleMessage is in this class
- Every helper method is moved to MAHService
- sendMessage is now called directly in the handleMessage
- DTO is now directly created in the handleMessage

RenaT0Pl committed:
refactor: Adjusted the expected HP for TowerImpl

Mostly from 2 to 9. Since we now calculate with 1 Tower witch has 10 hp.

RenaT0Pl committed:
Merge remote-tracking branch 'origin/25-monsterattackhandler' into 25-monsterattackhandler

# Conflicts:
#	src/test/java/at/aau/se2/service/MAHServiceTest.java

RenaT0Pl committed:
feat: Added two additional test cases for 100 % coverage

Added:
- testValidateEntities_InvalidNumberFormatForMonsterId,
- testValidateEntities_InvalidNumberFormatForTowerId
in MAHServiceTest
